configure.ac

```diff
@@ -424,9 +424,13 @@ AC_SEARCH_LIBS(
 # Lastly, take care of crosstool-NG internal values
 #--------------------------------------------------------------------
 # Hey! We need the date! :-)
-AC_SUBST(
-    [DATE],
-    [$(date +%Y%m%d)])
+DATE_FMT="%Y%m%d"
+if test "x$SOURCE_DATE_EPOCH" = "x"; then
+    DATE=$(date "+$DATE_FMT")
+else
+    DATE="$(date -u -d "@$SOURCE_DATE_EPOCH" "+$DATE_FMT" 2>/dev/null || date -u -r "$SOURCE_DATE_EPOCH" "+$DATE_FMT" 2>/dev/null || date -u "+$DATE_FMT")"
+fi
+AC_SUBST([DATE])
```

Comment:
Should be fine. According to the [autoconf manual](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Setting-Output-Variables.html#index-AC_005fSUBST-1203) this would substitute instances of `@DATE@` with the shell variable `$DATE`. The second form of `AC_SUBST([DATE], [$something])` is only needed because the variable names are different.

Still should actually be tested (which I haven't done).